src/Bridge/Doctrine/Orm/CollectionDataProvider.php

```diff
@@ -67,6 +68,9 @@ public function getCollection(string $resourceClass, string $operationName = nul
             }
         }
 
-        return $queryBuilder->getQuery()->getResult();
+        $query = $queryBuilder->getQuery();
+        //forces doctrine to not lazy load relations, see http://stackoverflow.com/questions/9848747/primary-key-of-owning-side-as-a-join-column
+        $query->setHint(Query::HINT_FORCE_PARTIAL_LOAD, true);
```

Comment:
@dunglas we may need to review this before merging I'm not sure that everyone would want this behavior. Wdyt?